python/demo/gsoc2022/demo_scattering_boundary_conditions.py

```diff
@@ -0,0 +1,448 @@
+# \mathbf{E}_b \cdot \bar{\mathbf{V}} d S \\ +& \int_{\partial \Omega}
+# \mathbf{n} \times \nabla \times \mathbf{E}_s \cdot \bar{\mathbf{V}}
+# +\left(j n_bk_{0}+\frac{1}{2r}\right) \mathbf{n} \times \mathbf{E}_s
+# \times \mathbf{n} \cdot \bar{\mathbf{V}} d l=0
+# \end{align}
+# $$
+#
+# By using the $(\nabla \times \mathbf{A}) \cdot \mathbf{B}=\mathbf{A}
+# \cdot(\nabla \times \mathbf{B})+\nabla \cdot(\mathbf{A}
+# \times \mathbf{B})$
+# relation, we can change the first term into:
+#
+# $$
+# \begin{align}
+# & \iint_{\Omega}-\nabla \cdot(\nabla\times\mathbf{E}_s \times
+# \bar{\mathbf{V}})-\nabla \times \mathbf{E}_s \cdot \nabla
+# \times\bar{\mathbf{V}}+\varepsilon_{r} k_{0}^{2} \mathbf{E}_s
+# \cdot \bar{\mathbf{V}}+k_{0}^{2}\left(\varepsilon_{r}-\varepsilon_b\right)
+# \mathbf{E}_b \cdot \bar{\mathbf{V}} d S \\ +&\int_{\partial \Omega}
+# \mathbf{n} \times \nabla \times \mathbf{E}_s \cdot \bar{\mathbf{V}}
+# +\left(j n_bk_{0}+\frac{1}{2r}\right) \mathbf{n} \times \mathbf{E}_s
+# \times \mathbf{n} \cdot \bar{\mathbf{V}} d l=0
+# \end{align}
+# $$
+#
+# using the divergence theorem $\iint_\Omega\nabla\cdot\mathbf{F}dS =
+# \int_{\partial\Omega} \mathbf{F}\cdot\mathbf{n}dl$, we can write:
+#
+# $$
+# \begin{align}
+# & \iint_{\Omega}-\nabla \times \mathbf{E}_s \cdot \nabla \times
+# \bar{\mathbf{V}}+\varepsilon_{r} k_{0}^{2} \mathbf{E}_s \cdot
+# \bar{\mathbf{V}}+k_{0}^{2}\left(\varepsilon_{r}-\varepsilon_b\right)
+# \mathbf{E}_b \cdot \bar{\mathbf{V}} d S \\ +&\int_{\partial \Omega}
+# -(\nabla\times\mathbf{E}_s \times \bar{\mathbf{V}})\cdot\mathbf{n}
+# + \mathbf{n} \times \nabla \times \mathbf{E}_s \cdot \bar{\mathbf{V}}
+# +\left(j n_bk_{0}+\frac{1}{2r}\right) \mathbf{n} \times \mathbf{E}_s
+# \times \mathbf{n} \cdot \bar{\mathbf{V}} d l=0
+# \end{align}
+# $$
+#
+# We can cancel $-(\nabla\times\mathbf{E}_s \times \bar{\mathbf{V}})
+# \cdot\mathbf{n}$  and $\mathbf{n} \times \nabla \times \mathbf{E}_s
+# \cdot \bar{\mathbf{V}}$ thanks to the triple product rule $\mathbf{A}
+# \cdot(\mathbf{B} \times \mathbf{C})=\mathbf{B} \cdot(\mathbf{C} \times
+# \mathbf{A})=\mathbf{C} \cdot(\mathbf{A} \times \mathbf{B})$, arriving
+# at the final weak form:
+#
+# $$
+# \begin{align}
+# & \iint_{\Omega}-\nabla \times \mathbf{E}_s \cdot \nabla \times
+# \bar{\mathbf{V}}+\varepsilon_{r} k_{0}^{2} \mathbf{E}_s \cdot
+# \bar{\mathbf{V}}+k_{0}^{2}\left(\varepsilon_{r}-\varepsilon_b\right)
+# \mathbf{E}_b \cdot \bar{\mathbf{V}} d S \\ +&\int_{\partial \Omega}
+# \left(j n_bk_{0}+\frac{1}{2r}\right) \mathbf{n} \times \mathbf{E}_s \times 
+# \mathbf{n} \cdot \bar{\mathbf{V}} d l=0
+# \end{align}
+# $$
+#
+# ## Implementation
+# The modules that will be used are imported:
+
+# +
+import numpy as np
+import ufl
+from gmsh_helpers import gmsh_model_to_mesh
+from mesh_wire import generate_mesh_wire
+from mpi4py import MPI
+from petsc4py import PETSc
+from scipy.constants import epsilon_0, mu_0
```

Comment:
```suggestion
from scipy.constants import epsilon_0, mu_0
```
Could we hardcode these constants instead, as it seems like overkill to requite scipy as a dependency of the demo